src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/IISUrlRewriteRule.cs

```diff
@@ -57,6 +65,22 @@ public virtual void ApplyRule(RewriteContext context)
                 }
             }
 
+            foreach (ServerVariable serverVariable in ServerVariables)
+            {
+                if (serverVariable.Name.StartsWith(RequestHeaderPrefix, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
Rather than doing string comparisons on the name, we can probably have an enum with different types (RequestHeaderPrefix, ResponseHeaderPrefix, etc.)